test/failed-files.test.js

```diff
@@ -151,5 +152,29 @@ describe('cosmiconfig', () => {
       expect(loadConfig(true)).toBe(null);
       return expect(loadConfig(false)).resolves.toBe(null);
     });
+
+    it('in sync mode, throws error if configPath is package.json and packageProp is false', () => {
+      expect(() =>
+        cosmiconfig('foo', { sync: true, packageProp: false }).load(
+          null,
+          path.join(__dirname, 'fixtures/package.json')
+        )
+      ).toThrow(/Please specify the packageProp option/);
+    });
+
+    it('in async mode, rejects if configPath is package.json and packageProp is false', () => {
+      return cosmiconfig('foo', { packageProp: false })
```

Comment:
If you add a `expect.assertions(1)`, there is no need to define the `.then` clause.